File: src/main/kotlin/me/alvr/asfui/views/ASF.kt
package me.alvr.asfui.views

import javafx.scene.Parent
import javafx.scene.control.Button
import javafx.scene.control.ComboBox
import me.alvr.asfui.ASFProcess
import me.alvr.asfui.util.Command
import me.alvr.asfui.util.OpenBrowser
import tornadofx.Fragment
import tornadofx.action
import tornadofx.enableWhen
import tornadofx.selectedItem

class ASF : Fragment() {
    override val root: Parent by fxml("/asf.fxml")
    private val openBrower: OpenBrowser by inject()

    private val asfHelp: Button by fxid("asf_help")
    private val asfRejoin: Button by fxid("asf_rejoin")
    private val asfUpdate: Button by fxid("asf_update")
    private val asfVersion: Button by fxid("asf_version")
    private val asfApi: Button by fxid("asf_api")

    private val bots: ComboBox<String> by param()
    private var bot = bots.selectedItem

    init {
        bots.valueProperty().addListener { _, _, newBot ->
            bot = newBot
        }

        asfHelp.apply {
            action {
                openBrower.openUrl("https://github.com/JustArchi/ArchiSteamFarm/wiki")
            }
            enableWhen(ASFProcess.started)
        }

        asfRejoin.apply {
            action {
                runAsync {
                    Command.sendCommand(Command.generateCommand(Command.REJOIN_CHAT, bot))
                }
            }
            enableWhen(ASFProcess.started)
        }

        asfUpdate.apply {
            runAsync {
                Command.sendCommand(Command.UPDATE)
            }
            enableWhen(ASFProcess.started)
        }

        asfVersion.apply {
            runAsync {
                Command.sendCommand(Command.VERSION)
            }
            enableWhen(ASFProcess.started)
        }

        asfApi.apply {
            runAsync {
                Command.sendCommand(Command.generateCommand(Command.API, bot))
            }
            enableWhen(ASFProcess.started)
        }
    }
}

File: src/views/asf.py
import threading
import webbrowser

from PyQt5 import uic
from PyQt5.QtWidgets import QWidget

from utils.command import generate_command, send_command, REJOIN, UPDATE, VERSION, API
from utils.resources import resource_path


class ASF(QWidget):
    def __init__(self, bots):
        super().__init__()

        uic.loadUi(resource_path('src/resources/ui/asf.ui'), self)

        self.bots = bots
        self.asf_help.clicked.connect(self.help)
        self.asf_rejoin.clicked.connect(self.rejoin)
        self.asf_update.clicked.connect(self.up)
        self.asf_version.clicked.connect(self.version)
        self.asf_api.clicked.connect(self.api)

    def help(self):
        webbrowser.open_new_tab('https://github.com/JustArchi/ArchiSteamFarm/wiki')

    def rejoin(self):
        command = generate_command(REJOIN, self.bots.currentText())
        threading.Thread(target=send_command, args=(command,), daemon=True).start()

    def up(self):
        threading.Thread(target=send_command, args=(UPDATE,), daemon=True).start()

    def version(self):
        threading.Thread(target=send_command, args=(VERSION,), daemon=True).start()

    def api(self):
        command = generate_command(API, self.bots.currentText())
        threading.Thread(target=send_command, args=(command,), daemon=True).start()
